go/signedexchange/cmd/dump-signedexchange/main.go

```diff
@@ -5,50 +5,81 @@ import (
 	"fmt"
 	"io/ioutil"
 	"log"
+	"net/http"
 	"os"
 	"time"
 
 	"github.com/WICG/webpackage/go/signedexchange"
 )
 
 var (
-	flagInput     = flag.String("i", "", "Signed-exchange input file")
-	flagSignature = flag.Bool("signature", false, "Print only signature value")
-	flagVerify    = flag.Bool("verify", false, "Perform signature verification")
 	flagCert      = flag.String("cert", "", "Certificate CBOR file. If specified, used instead of fetching from signature's cert-url")
+	flagHeaders   = flag.Bool("headers", true, "Print headers")
+	flagFilename  = flag.String("i", "", "Signed-exchange input file")
+	flagPayload   = flag.Bool("payload", true, "Print payload")
+	flagSignature = flag.Bool("signature", false, "Prints only signature value")
+	flagURI       = flag.String("uri", "", "Signed-exchange uri")
+	flagVerify    = flag.Bool("verify", false, "Perform signature verification")
 )
 
 func run() error {
-	in := os.Stdin
-	if *flagInput != "" {
-		var err error
-		in, err = os.Open(*flagInput)
+	fi, err := os.Stdin.Stat()
+	var e *signedexchange.Exchange
+	if *flagFilename != "" {
+		in, err := os.Open(*flagFilename)
 		if err != nil {
 			return err
 		}
 		defer in.Close()
```

Comment:
These four lines are the same in each branch of the if. Maybe just `var in` outside the `if` and set it in each branch.